GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -459,25 +459,6 @@ private void _indexWatcher_Changed(object sender, IndexChangedEventArgs e)
             });
         }
 
-        private bool _pluginsLoaded;
-        private void LoadPluginsInPluginMenu()
-        {
-            if (_pluginsLoaded)
```

Comment:
@gerhardol The bad state we observed was caused by this condition. It should have been a condition similar to the following:

```
if (_pluginsLoaded || !PluginLoader.HasLoadedPlugins)
{
    return;
}
```

I ended up changing the "populate on demand" to a "populate when ready" path.